frontend/src/Settings/Quality/Definition/QualityDefinition.js

```diff
@@ -48,20 +36,29 @@ class QualityDefinition extends Component {
   constructor(props, context) {
     super(props, context);
 
-    this._forceUpdateTimeout = null;
+    const { sizeLimits } = this.props;
 
-    this.state = {
-      sliderMinSize: getSliderValue(props.minSize, slider.min),
-      sliderMaxSize: getSliderValue(props.maxSize, slider.max),
-      sliderPreferredSize: getSliderValue(props.preferredSize, (slider.max - 3))
-    };
+    sizeLimits.clampMax = (inputValue) =>
```

Comment:
`sizeLimits` comes from the redux store and this mutates that state. Mutating the store outside of an action in a way that will break and is not reactive.

These should be instance functions instead.

Reply:
Could you provide an example, for my educational benefit and to ensure I make the changes you're expecting?

Reply:
Since you need to reference `this.props` in these functions you'll want to do something similar to how `getSelectedIds` is implemented here:

https://github.com/Sonarr/Sonarr/blob/14005d8d1054eafaba808337a109d5812f3e79e6/frontend/src/Wanted/CutoffUnmet/CutoffUnmet.js#L65-L67

```
clampMaxAndTranslateNull = (inputValue) => {
	const { minLimit, maxLimit } = this.props.sizeLimits;

   return clampRangeAndTranslateNull(inputValue, sizeLimits.minLimit, sizeLimits.maxLimit);
}

the caller of the functions would call `this.clampMaxAndTranslateNull(inputValue)`

Reply:
I have reverted all frontend changes and this will be resolved in my next push.